src/main/kotlin/net/lumalyte/lg/infrastructure/services/WarServiceBukkit.kt

```
@@ -327,10 +302,102 @@ class WarServiceBukkit(
+        if (loserGuildId != null && combat.warLoseExperience > 0) {
+            awardGuildExperience(loserGuildId, combat.warLoseExperience)
+        }
+    }
+
+    private fun awardGuildExperience(guildId: UUID, amount: Int) {
+        try {
+            val progression = progressionRepository.getGuildProgression(guildId) ?: return
+            progressionRepository.saveGuildProgression(
+                progression.copy(totalExperience = progression.totalExperience + amount)
+            )
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)